Jack/Shrink.fs

```diff
@@ -108,16 +108,15 @@ module Shrink =
             LazyList.consNub destination <|
             LazyList.map (fun y -> x - y) (halves diff)
 
-    /// Shrink a floating point number.
-    let double (x : double) : LazyList<double> =
+    /// Shrink a number.
+    let inline number x : LazyList<'a> =
         let positive =
-            if x < 0.0 then
+            if x < LanguagePrimitives.GenericZero then
                 LazyList.singleton (-x)
             else
                 LazyList.empty
 
         let integrals =
-            towards 0I (bigint x)
-            |> LazyList.map double
+            towards LanguagePrimitives.GenericZero x
```

Comment:
So, the old code was deliberately forcing the value to an integer (which is what the current version of QuickCheck does in Haskell now), because I think floating point math can put us in a situation where shrinking doesn't terminate.